server/lib/pbench/__init__.py

```diff
@@ -153,7 +153,8 @@ def get_pbench_logger(caller, config):
         except FileExistsError:
             # directory already exists, ignore
             pass
-        fh = logging.FileHandler(os.path.join(logdir, "{}.log".format(caller)))
+        #fh = logging.FileHandler(os.path.join(logdir, "{}.log".format(caller)))
+        fh = SysLogHandler(address='/dev/log', facility=6, socktype = socket.SOCK_DGRAM)
```

Comment:
We need options here, something like:

```
    if config.logger_type == "file":
        handler = logging.FileHandler(os.path.join(logdir, "{}.log".format(caller)))
    elif config.logger_type == "devlog":
        handler = logging.SyslogHandler(address='/dev/log', facility=6)
    elif config.logger_type == "hostport":
        handler = logging.SysLogHandler(address=(config.logger_host, config.logger_port), facility=6, socktype = socket.SOCK_DGRAM)
    else:
        raise Exception("Unsupported logger type")
```

And change the references to `fh` to `handler` since we no longer always have a "(f)ile (h)andler".

Then change the `PbenchConfig()` class to look for the "`logger_type`", defaulting it to "`devlog`".

Reply:
@portante Please may I know. Why do we need different logging Handlers? We were supposed to have FileHandler for unittests and SysLogHandler for the rest.

Reply:
I believe it's for flexibility: /dev/log allows logging to the rsyslog running on the server (which may in turn decide to send the logs to a remote server), whereas the logger_host/logger_port allows us to bypass the rsyslog on the server and send directly to a remote rsyslog server.

Reply:
So, I understand that we need to distinguish between the three types of files. Just wanted to ask if there is a predefined way in pbench to do this or we need to create this functionality. Also, if we are going to create this then is there going to be a specific way to create this or we can do this using a hack.
I tried to find a way to get log type using:
- I have been through log files but couldn't find anything that will specify the type of log files, not even in the log messages.
- tried to find any function or attribute inside PbenchConfig but no help there.